lib/html/static/js/fileSearch.js

```diff
@@ -0,0 +1,105 @@
+/******************************************************************************
+ * (C) British crown copyright 2012-6 Met Office.
+ *
+ * This file is part of Rose, a framework for scientific suites.
+ *
+ * Rose is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Rose is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Rose. If not, see <http://www.gnu.org/licenses/>.
+ *
+ ******************************************************************************/
+var FileLoader = ( function() {
+    // wrapper to highlight the selected line
+    var mark = ['<span class="highlight">', '</span>'];
+
+    var loadFile = function(search_string, goto) {
+		/* Loads in a file from the provided url, populates the line
+		 * numbers and scrolls to the goto line is provided. */
+		
+		// detect search mode
+		search_mode = $("input:radio[name=search_mode]:checked").val();
+		// generate the search url
+        url = getCleanURL() + "&search_string=" + search_string +
+            "&search_mode=" + search_mode;
+		url = url.replace('/view/', '/file_view/');
```

Comment:
The url parameters for file_view are exactly the same as for view only with the addition of search_string and search_mode.
/<b>view</b>/user/suite?path=/path/to/file

results in an ajax call to:

/<b>file_view</b>/user/suite?path=/path/to/file?search_string="hello"&search_mode="TEXT"

I can dump these parameters into hidden form elements then reconstruct the url from the parameters if you like.